### ISSUE TITLE
Check activation of Pod Security Standards

### ISSUE THREAD

**Issue by TorstenD-SAP** [kyma-project/kyma#15897]
**Description**

We have to check, whether our Pod specifications meat the requirements of the Pod Security Standards or not. I created a [document describing the problem and listing a couple of steps to perform](https://github.com/kyma-project/community/tree/main/concepts/psp-replacement) in order to find out if we are able to activate Pod Security Standards in which level. The overall goal should be to enforce the Pod Security Standard _restricted_ on the Kyma namespace.

The [steps](https://github.com/kyma-project/community/tree/main/concepts/psp-replacement#compliance-with-pod-security-standards) listed in the document mentioned above must be performed and the results must be added to this issue.

**Reasons**

Pod Security Policies is no longer part of Kubernetes in version 1.25. As a replacement, Kubernetes introduced the so-called [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards) and an admission controller ([Pod Security Admission](https://kubernetes.io/docs/concepts/security/pod-security-admission/)) to enforce the Pod Security Standards.

To not wait until Pod Security Policies are finally removed, this should be done as part of the upcoming Kubernetes update.
See #15732 Kubernetes upgrade from ver 1.23 to 1.24

**Attachments**

**Comment by a-thaler** [comment]
We will have a problem with Daemonsets which typically require access to the node, see also my [PR](https://github.com/kyma-project/kyma/pull/15744) for the observability area which has two daemonsets

**Comment by pPrecel** [comment]
The output from Otters' investigation is described [here](https://github.com/kyma-project/kyma/issues/15245#issuecomment-1293215929).

**Comment by anoipm** [comment]
Function-controller (serverless) in kyma was adapted to be compatible with PSS. Now we have some incompabilities described [here](https://github.com/kyma-project/kyma/issues/15923#issuecomment-1310140557)

**Comment by dariusztutaj** [comment]
Istio with CNI plugin enabled has one container which needs to be run with uid 0. This container is not directly attached to the workloads. More info [here](https://github.com/kyma-project/kyma/issues/15916#issuecomment-1316511519)

**Comment by marcobebway** [comment]
Here is the applied security contexts config for:
- [Eventing](https://github.com/kyma-project/kyma/issues/15919#issuecomment-1412352214).
- [KMC](https://github.com/kyma-project/kyma/issues/15919#issuecomment-1412354351).

**Comment by a-thaler** [comment]
In the linked PR https://github.com/kyma-project/kyma/issues/15919#issuecomment-1412352214 I saw that the NET_BIND_SERVICE capability got added. @marcobebway does your workload really require that capability? I think that is a leftover of the istio-proxy requirements which are not needed anymore with the CNI plugin active. @TorstenD-SAP could you verify that please and make that more explicit in the guidance: https://github.com/kyma-project/community/blob/main/concepts/psp-replacement/restricted-pod-spec.yaml

**Comment by TorstenD-SAP** [comment]
The resources provided by Kyma fulfill different standards (some of them are even not compliant with baseline):

| Kyma component | K8s kind | K8s resource name | Namespace | Status | Minimal compliant standard |
|---|---|---|---|---|---|
| API Exposure | Job | api-gateway-init | kyma-system | done | baseline |
| Connectivity | Deployment | central-application-connectivity-validator | kyma-system | done | baseline |
| Connectivity | Deployment | central-application-gateway | kyma-system | done | baseline |
| Eventing | Statefulset | eventing-nats | kyma-system | done | baseline |
| SRE | Deployment | rma-kube-state-metrics | kyma-system | done | baseline |
| SRE | Deployment | rma-victoria-metrics-agent | kyma-system | done | baseline |
| Connectivity | Deployment | sap-btp-operator-controller-manager | kyma-system | done | baseline |
| Serverless | Deployment | serverless-docker-registry | kyma-system | done | baseline |
| Observability | Deployment | tracing-jaeger | kyma-system | done | baseline |
| Security | Deployment | warden-admission | kyma-system | done | baseline |
| Security | Deployment | warden-operator | kyma-system | done | baseline |
| API Exposure | Cronjob | api-gateway-certifates-cronjob | kyma-system | done | none |
| Eventing | Job | eventing-controller-cert-handler | kyma-system | done | none |
| API Exposure | Daemonset | istio-cni-node | istio-system | done | none |
| Observability | Daemonset | monitoring-prometheus-node-exporter | kyma-system | done | none |
| API Exposure | Cronjob | oathkeeper-jwks-rotator | kyma-system | done | none |
| Serverless | Daemonset | serverless-docker-registry-self-signed-cert | kyma-system | done | none |
| Observability | Daemonset | telemetry-fluent-bit | kyma-system | done | none |
| Observability | Statefulset | alertmanager-monitoring-alertmanager | kyma-system | done | restricted |
| API Exposure | Deployment | api-gateway | kyma-system | done | restricted |
| Connectivity | Deployment | compass-runtime-agent | kyma-system | done | restricted |
| Eventing | Deployment | eventing-controller | kyma-system | done | restricted |
| Eventing | Deployment | eventing-publisher-proxy | kyma-system | done | restricted |
| API Exposure | Deployment | istiod | istio-system | done | restricted |
| API Exposure | Deployment | istio-ingressgateway | istio-system | done | restricted |
| Observability | Statefulset | logging-loki | kyma-system | done | restricted |
| Observability | Deployment | monitoring-auth-proxy-grafana | kyma-system | done | restricted |
| Observability | Deployment | monitoring-grafana | kyma-system | done | restricted |
| Observability | Deployment | monitoring-kube-state-metrics | kyma-system | done | restricted |
| Observability | Deployment | monitoring-operator | kyma-system | done | restricted |
| Observability | Deployment | monitoring-prometheus-istio-server | kyma-system | done | restricted |
| API Exposure | Deployment | ory-hydra | kyma-system | done | restricted |
| API Exposure | Deployment | ory-hydra-maester | kyma-system | done | restricted |
| API Exposure | Deployment | ory-oathkeeper | kyma-system | done | restricted |
| API Exposure | Pod | ory-postgresql | kyma-system | done | restricted |
| Observability | Statefulset | prometheus-monitoring-prometheus | kyma-system | done | restricted |
| Serverless | Deployment | serverless-ctrl-mngr | kyma-system | done | restricted |
| Serverless | Deployment | serverless-webhook-svc | kyma-system | done | restricted |
| Observability | Deployment | telemetry-operator | kyma-system | done | restricted |
| Observability | Deployment | telemetry-trace-collector | kyma-system | done | restricted |
| Observability | Deployment | tracing-auth-proxy | kyma-system | done | restricted |
| Observability | Deployment | tracing-jaeger-operator | kyma-system | done | restricted |

**Comment by TorstenD-SAP** [comment]
After all the teams have finalized, it turned out that some of the Kyma related resource even not fulfill the baseline standard.

| Kyma component | K8s kind | K8s resource name | Namespace | Status | Minimal compliant standard |
|---|---|---|---|---|---|
| API Exposure | Job | api-gateway-init | kyma-system | done | baseline |
| Connectivity | Deployment | central-application-connectivity-validator | kyma-system | done | baseline |
| Connectivity | Deployment | central-application-gateway | kyma-system | done | baseline |
| Eventing | Statefulset | eventing-nats | kyma-system | done | baseline |
| SRE | Deployment | rma-kube-state-metrics | kyma-system | done | baseline |
| SRE | Deployment | rma-victoria-metrics-agent | kyma-system | done | baseline |
| Connectivity | Deployment | sap-btp-operator-controller-manager | kyma-system | done | baseline |
| Serverless | Deployment | serverless-docker-registry | kyma-system | done | baseline |
| Observability | Deployment | tracing-jaeger | kyma-system | done | baseline |
| Security | Deployment | warden-admission | kyma-system | done | baseline |
| Security | Deployment | warden-operator | kyma-system | done | baseline |
| API Exposure | Cronjob | api-gateway-certifates-cronjob | kyma-system | done | none |
| Eventing | Job | eventing-controller-cert-handler | kyma-system | done | none |
| API Exposure | Daemonset | istio-cni-node | istio-system | done | none |
| Observability | Daemonset | monitoring-prometheus-node-exporter | kyma-system | done | none |
| API Exposure | Cronjob | oathkeeper-jwks-rotator | kyma-system | done | none |
| Serverless | Daemonset | serverless-docker-registry-self-signed-cert | kyma-system | done | none |
| Observability | Daemonset | telemetry-fluent-bit | kyma-system | done | none |
| Observability | Statefulset | alertmanager-monitoring-alertmanager | kyma-system | done | restricted |
| API Exposure | Deployment | api-gateway | kyma-system | done | restricted |
| Connectivity | Deployment | compass-runtime-agent | kyma-system | done | restricted |
| Eventing | Deployment | eventing-controller | kyma-system | done | restricted |
| Eventing | Deployment | eventing-publisher-proxy | kyma-system | done | restricted |
| API Exposure | Deployment | istiod | istio-system | done | restricted |
| API Exposure | Deployment | istio-ingressgateway | istio-system | done | restricted |
| Observability | Statefulset | logging-loki | kyma-system | done | restricted |
| Observability | Deployment | monitoring-auth-proxy-grafana | kyma-system | done | restricted |
| Observability | Deployment | monitoring-grafana | kyma-system | done | restricted |
| Observability | Deployment | monitoring-kube-state-metrics | kyma-system | done | restricted |
| Observability | Deployment | monitoring-operator | kyma-system | done | restricted |
| Observability | Deployment | monitoring-prometheus-istio-server | kyma-system | done | restricted |
| API Exposure | Deployment | ory-hydra | kyma-system | done | restricted |
| API Exposure | Deployment | ory-hydra-maester | kyma-system | done | restricted |
| API Exposure | Deployment | ory-oathkeeper | kyma-system | done | restricted |
| API Exposure | Pod | ory-postgresql | kyma-system | done | restricted |
| Observability | Statefulset | prometheus-monitoring-prometheus | kyma-system | done | restricted |
| Serverless | Deployment | serverless-ctrl-mngr | kyma-system | done | restricted |
| Serverless | Deployment | serverless-webhook-svc | kyma-system | done | restricted |
| Observability | Deployment | telemetry-operator | kyma-system | done | restricted |
| Observability | Deployment | telemetry-trace-collector | kyma-system | done | restricted |
| Observability | Deployment | tracing-auth-proxy | kyma-system | done | restricted |
| Observability | Deployment | tracing-jaeger-operator | kyma-system | done | restricted |

The Pod Security Standards can only be applied on Namespace level and the definition of exceptions of is very limited. The option to except by user is not possible because the deployment is done via automation. The exception by runtimeClassName is not possible because our clusters have only on runtime class. The only option which could be used is the separation by Namespace. This option should be discussed.

Since the Pod Security Standards in combination with Pod Security Admission is very inflexible, I suggest to not activate Pod Security Admission. I would rather recommend to support our developers how to develop secure K8s manifests and check their manifests against security best practices. Probably we find a team which is interested in developing a module containing a policy enforcement tool like OPA Gatekeeper.

**Comment by TorstenD-SAP** [comment]
We decided today to not activate Pod Security Admission because several Kyma-related resources are not compliant and the only way to separate is to have individual namespaces. We also decided to go forward with the already started approach to secure workloads as much as possible. This should be supported be appropriate tooling which allows developers to check whether their specifications are secure or not.
